Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -305,6 +305,9 @@ auto TreeResolver::resolveElement(Element& element, const RenderStyle* existingS
         }
     }
 
+    if (update.style->display() == DisplayType::None)
+        return { WTFMove(update), descendantsToResolve };
```

Comment:
If this early return is intended to just avoid pseudo-element related work, then this way of doing it is error-prone, because people may add code after here which is not pseudo-related. In fact, there's already touch-action and userModify code at the end of the function that this will skip, and it's not clear if that's OK.

If you really want to just skip the pseudo-related work, move that into its own function and explicitly skip it for display:none.

Reply:
> If this early return is intended to just avoid pseudo-element related work, then this way of doing it is error-prone, because people may add code after here which is not pseudo-related. In fact, there's already touch-action and userModify code at the end of the function that this will skip, and it's not clear if that's OK.

I think it's correct to skip setting those two document flags, hidden editable elements aren't usable to the user, and ditto for elements with non-auto touch actions.

I'll A/B test to see how much improvement this makes.